### PR TITLE
Task/notas

### PR DESCRIPTION
**Pull request by Icaro-Lima**

### CODE REVIEW

**Comment by paulofelipefeitosa** [comment]
This pull request **introduces 3 alerts** when merging 128496b7180da54bbf3690b0128bcf3658d5c340 into 71a837a4183d68ff69985e408e4d35955da6b1ef - [view on LGTM.com](https://lgtm.com/projects/g/OpenEnade/API/rev/pr-4f6957a2cfa9cb5af4f7f8aad14f686c87007b81)

**new alerts:**

* 3 for Dereferenced variable may be null

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by paulofelipefeitosa** [comment]
This pull request **introduces 3 alerts** when merging b180cfcfd64603e4494ef2de34120140c3d52384 into 71a837a4183d68ff69985e408e4d35955da6b1ef - [view on LGTM.com](https://lgtm.com/projects/g/OpenEnade/API/rev/pr-c84b37a3d26482a414cc1cf53f0251a72e2f3f7f)

**new alerts:**

* 3 for Dereferenced variable may be null

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by paulofelipefeitosa** [comment]
This pull request **introduces 3 alerts** when merging d776ab863d62da6c2128187c0e9f90435830e36f into 71a837a4183d68ff69985e408e4d35955da6b1ef - [view on LGTM.com](https://lgtm.com/projects/g/OpenEnade/API/rev/pr-1a562b9647ce5133833016cd9e2402f4360a14b7)

**new alerts:**

* 3 for Dereferenced variable may be null

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/main/java/br/com/openenade/api/curso/CursoId.java [hunk 1]
```diff
@@ -24,4 +24,20 @@ public CursoId(Long codigoCurso, Modalidade modalidade) {
         this.modalidade = modalidade;
     }
 
+    public Long getCodigoCurso() {
```
Comment (paulofelipefeitosa) [inline src/main/java/br/com/openenade/api/curso/CursoId.java]: I think that these methods in CursoId are not necessary.

File: src/main/java/br/com/openenade/api/nota/Avaliacao.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    private int concluintesParticipantes;
+    private double notaBrutaFG;
+    private double notaPadronizadaFG;
+    private double notaBrutaCE;
+    private double notaPadronizadaCE;
+    private double enadeContinuo;
+    private int enadeFaixa;
+
+    @Override
+    public int hashCode() {
```
Comment (paulofelipefeitosa) [inline src/main/java/br/com/openenade/api/nota/Avaliacao.java]: No need for hashCode and equals methods in Avaliacao.

File: src/main/java/br/com/openenade/api/nota/NotaController.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	public ResponseEntity<List<Nota>> getAll() {
+		return new ResponseEntity<>(this.service.getAll(), HttpStatus.OK);
+	}
+
+	@ResponseBody
+	@GetMapping(path = "/{ano}-{codigoCurso}-{modalidade}-{codigoIES}-{codigoMunicipio}")
+	public ResponseEntity<Nota> getNotaByIndex(NotaIdInterface notaIdInterface) {
+
+		Optional<Nota> optNota = null;
+		try {
```
Comment (paulofelipefeitosa) [inline src/main/java/br/com/openenade/api/nota/NotaController.java]: I think that this business logic should be implemented in NotaService. Therefore, there is a ResourceNotFound Exception that already results in an HTTP.NOT_FOUND status with a message, EstadoController use it.

File: src/main/java/br/com/openenade/api/nota/NotaId.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		NotaId other = (NotaId) obj;
+
+		boolean anoCond = ano == null && other.ano == null || ano.equals(other.ano);
```
Comment (paulofelipefeitosa) [inline src/main/java/br/com/openenade/api/nota/NotaId.java]: LGTM is reporting some warnings about these conditions, don't worry about the bots reporting some unrealistic problems.

File: src/main/java/br/com/openenade/api/nota/NotaService.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Autowired
+    private CursoRepository cursoRepository;
+
+    @Autowired
+    private UniversidadeRepository universidadeRepository;
+
+    public Nota save(Nota nota) {
+
+        this.anoRepository.save(nota.getInfo().getAno());
```
Comment (paulofelipefeitosa) [inline src/main/java/br/com/openenade/api/nota/NotaService.java]: I think that is not good to overwrite Curso and Universidade in the database applying consecutive saves for each Nota insertion.

File: src/main/java/br/com/openenade/api/nota/NotaService.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        Optional<Universidade> optUniversidade =
+                this.universidadeRepository.findById(universidadeId);
+
+        NotaId notaId = new NotaId(optAno.get(), optCurso.get(), optUniversidade.get());
+
+        return this.notaRepository.findById(notaId);
+    }
+
+    public boolean deleteNotaById(NotaId id) {
+        if (this.notaRepository.existsById(id)) {
```
Comment (paulofelipefeitosa) [inline src/main/java/br/com/openenade/api/nota/NotaService.java]: There is no need for this condition if the Nota is not in the database the return should be a NotFound HTTP response status, you just apply the operation (see EstadoService)

**Comment by paulofelipefeitosa** [comment]
This pull request **introduces 3 alerts** when merging fa5a57d77453b01ed6cc38c6439e00ad6d98e268 into 71a837a4183d68ff69985e408e4d35955da6b1ef - [view on LGTM.com](https://lgtm.com/projects/g/OpenEnade/API/rev/pr-8f14bf538531b25d28be77d67a6ce2a4e2dae3f3)

**new alerts:**

* 3 for Dereferenced variable may be null

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/main/java/br/com/openenade/api/curso/CursoId.java (hunk 1 above)
Comment (Icaro-Lima) [reply src/main/java/br/com/openenade/api/curso/CursoId.java]: It is for remove certain warnings.

File: src/main/java/br/com/openenade/api/nota/Avaliacao.java (hunk 2 above)
Comment (Icaro-Lima) [reply src/main/java/br/com/openenade/api/nota/Avaliacao.java]: Ok o/

File: src/main/java/br/com/openenade/api/nota/NotaId.java (hunk 4 above)
Comment (Icaro-Lima) [reply src/main/java/br/com/openenade/api/nota/NotaId.java]: I did this just to remove some Codacy warnings

File: src/main/java/br/com/openenade/api/nota/NotaService.java (hunk 5 above)
Comment (Icaro-Lima) [reply src/main/java/br/com/openenade/api/nota/NotaService.java]: It was only to short, we will fix it in the future.

File: src/main/java/br/com/openenade/api/nota/NotaService.java (hunk 6 above)
Comment (Icaro-Lima) [reply src/main/java/br/com/openenade/api/nota/NotaService.java]: Ok!

File: src/main/java/br/com/openenade/api/nota/NotaId.java (hunk 4 above)
Comment (paulofelipefeitosa) [reply src/main/java/br/com/openenade/api/nota/NotaId.java]: Do not worry about some unrealistic problems reported by the bots.

File: src/main/java/br/com/openenade/api/curso/CursoId.java (hunk 1 above)
Comment (paulofelipefeitosa) [reply src/main/java/br/com/openenade/api/curso/CursoId.java]: Ok.

**Comment by paulofelipefeitosa** [comment]
This pull request **introduces 3 alerts** when merging a20081a3f870096d7eb85e1afa92c67575285bbd into 71a837a4183d68ff69985e408e4d35955da6b1ef - [view on LGTM.com](https://lgtm.com/projects/g/OpenEnade/API/rev/pr-f553b10158df28430ba99f96a023903d3e3bd79a)

**new alerts:**

* 3 for Dereferenced variable may be null

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/main/java/br/com/openenade/api/nota/NotaId.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((ano == null) ? 0 : ano.hashCode());
+        result = prime * result + ((curso == null) ? 0 : curso.hashCode());
+        result = prime * result + ((universidade == null) ? 0 : universidade.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```
[the remark this answers is not in the document]
Comment (Icaro-Lima) [reply src/main/java/br/com/openenade/api/nota/NotaId.java]: Já comentamos sobre.

File: src/main/java/br/com/openenade/api/nota/NotaService.java (hunk 5 above)
Comment (paulofelipefeitosa) [reply src/main/java/br/com/openenade/api/nota/NotaService.java]: @Icaro-Lima this issue is not resolved.

**Pull request merged**